File: zlzlelzl/leetcode/Linked List/Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        stack = []
        tail = ListNode()
        temp_tail = tail

        while head:
            stack.append(head)
            head = head.next

        while stack:
            stacked = stack.pop()
            tail.next = ListNode()
            tail.next.val = stacked.val
            tail = tail.next

        return temp_tail.next

File: JunnyJun/Reverse Linked List.cpp
class Solution
{
public:
    ListNode *reverseList(ListNode *head)
    {
        ListNode *prev = NULL;

        while (head)
        {
            ListNode *tail = head->next;
            head->next = prev;
            prev = head;
            head = tail;
        }
        return prev;
    }
};
